src/components/screens/timebank/timebank-screen.tsx

```diff
@@ -0,0 +1,119 @@
+  const [personTotalTime, setPersonTotalTime] = useAtom(personTotalTimeAtom);
+  const [personDailyEntry, setPersonDailyEntry] = useAtom(personDailyEntryAtom);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+
+  /**
+   * Gets the logged in person from the persons list atom and then gets that person's total time data.
+   * Function is called on initial load / persons list changes or timespan changes.
+   * @param timespan Optional Timespan string which controls whether @PersonTotalTime results are condensed into weeks, months, years or all time. Defaults to all time.
+   */
+  const getPersonTotalTime = async (timespan?: Timespan): Promise<void> => {
+    if (!personTotalTime || timespan) {
+      setPersonTotalTimeLoading(true);
+      setTimespanSelector(timespan || Timespan.ALL_TIME);
+      if (persons.length) {
+        try {
+          const loggedInPerson = persons.filter(
+            (person: Person) => person.keycloakId === config.keycloak.id || userProfile?.id
+          )[0];
+          const fetchedPersonTotalTime = await personsApi.listPersonTotalTime({
+            personId: loggedInPerson.id,
+            timespan: timespan
+          });
+          setPersonTotalTime(fetchedPersonTotalTime[0]);
+        } catch (error) {
+          setError(`${strings.error.totalTimeFetch}, ${error}`);
+        }
+        setPersonTotalTimeLoading(false);
+      }
+    }
+  };
+
+  /**
+   * Gets the logged in person from persons list atom and then fetches the person's daily entries.
+   */
+  const getPersonDailyEntries = async (): Promise<void> => {
+    if (persons.length) {
+      try {
+        const loggedInPerson = persons.filter(
+          (person: Person) => person.keycloakId === config.keycloak.id || userProfile?.id
+        )[0];
+        const fetchedDailyEntries = await dailyEntriesApi.listDailyEntries({
+          personId: loggedInPerson.id
+        });
+        setDailyEntries(fetchedDailyEntries);
+        setPersonDailyEntry(fetchedDailyEntries[0]);
+      } catch (error) {
+        setError(`${strings.error.dailyEntriesFetch}, ${error}`);
+      }
+      setPersonTotalTimeLoading(false);
+    }
+  };
+
+  /**
+   * Changes the displayed daily entry in the pie chart via the Date Picker.
+   * @param selectedDate selected date from DatePicker
+   */
+  const handleDailyEntryChange = (selectedDate: DateTime | null) => {
+    setPersonDailyEntry(
+      dailyEntries.filter(
+        (item) => DateTime.fromJSDate(item.date).toISODate() === selectedDate?.toISODate()
+      )[0]
+    );
+  };
+
+  useEffect(() => {
+    getPersonTotalTime();
+    getPersonDailyEntries();
+  }, [persons]);
+
+  if (personTotalTimeLoading || !personDailyEntry || !dailyEntries || !personTotalTime)
+    return (
+      <Card sx={{ p: "25%", display: "flex", justifyContent: "center" }}>
+        <CircularProgress sx={{ scale: "150%" }} />
+      </Card>
+    );
+  else if (personDailyEntry && dailyEntries && personTotalTime)
```

Comment:
These conditions should be updated so that the else if is not required. Perhaps you could create a new variable containing the boolean values you are using in the conditional to make it a little cleaner.
IS it also definitely the case you want to return a loading spinner if the cases !personTotalTime, !dailyEntries, !personDailyEntry?